src/Mvc/Mvc.Api.Analyzers/src/ActualApiResponseMetadataFactory.cs

```diff
@@ -102,13 +102,13 @@ public static class ActualApiResponseMetadataFactory
                 case ObjectCreationExpressionSyntax creation:
                     {
                         // Read values from 'return new StatusCodeResult(200) case.
-                        var result = InspectMethodArguments(semanticModel, creation, creation.ArgumentList, cancellationToken);
+                        var result = InspectMethodArguments(semanticModel, creation, creation.ArgumentList!, cancellationToken);
```

Comment:
`ObjectCreationExpressionSyntax.ArgumentList` is nullable in roslyn: http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Generated/Syntax.xml.Syntax.Generated.cs,2849

Should you instead update `InspectMethodArguments` to take a nullable argument list and check for null before accessing it? How do you know/guarantee that ArgumentList will not be null here?

Reply:
@NTaylorMullen @pranavkm seems like this is a question that's best answered by folks familiar with Mvc.Api.Analyzers.

I mostly updated the code to replicate the previous behaviour: if it didn't handle nulls, I assume it's because the value is expected to be not null. Ditto for the other questions pointed out by @safern

Reply:
It might not hurt to add a null check. That said, this is inspecting MVC's types, which do not have a parameterless ctor so far.

Reply:
It's pretty much the theme with the other ones. While adding a null check would certainly not hurt, most of these APIs are introspecting MVC's code. Some of it is specifically annotated for these analyzers to function correctly. We would have to do done something wrong in our code for these to null-ref.